src/markdown.cpp

```diff
@@ -972,17 +972,20 @@ int Markdown::processLink(const char *data,int,int size)
   if (!isImageLink && content.isEmpty()) { TRACE_RESULT(0); return 0; } // no link text
   i++; // skip over ]
 
+  bool whiteSpace = false;
   // skip whitespace
-  while (i<size && data[i]==' ') i++;
+  while (i<size && data[i]==' ') {whiteSpace = true; i++;}
   if (i<size && data[i]=='\n') // one newline allowed here
   {
+    whiteSpace = true;
     i++;
     nl++;
     // skip more whitespace
     while (i<size && data[i]==' ') i++;
   }
   nlTotal += nl;
   nl = 0;
+  if (whiteSpace && i<size && (data[i]=='(' || data[i]=='[')) return 0;
```

Comment:
It looks like a space is not allowed for an inline link but it is for reference link, see https://www.markdownguide.org/basic-syntax/#reference-style-links, so I think check for `data[i]=='['` should be removed. To complicate things a newline is not allowed, so probably spaces and newlines need to be tracked separately.

Reply:
I think this depends a bit on what the outcome should be. When having:
```
Square bracket
[hobbit-hole][1]

Square bracket, now with space
[hobbit-hole] [1]

End of text

[1]: https://en.wikipedia.org/wiki/Hobbit#Lifestyle
```
in the implementations on:
- https://spec.commonmark.org/dingus/
- https://markdownlivepreview.com/
- Here on GitHub:
   Square bracket
   [hobbit-hole][1]

   Square bracket, now with space
   [hobbit-hole] [1]

   End of text

   [1]: https://en.wikipedia.org/wiki/Hobbit#Lifestyle
- stackexchange

we get (I made the link red for clarity):

![image](https://user-images.githubusercontent.com/5223533/144588518-23c86484-499c-4f5e-996f-40a39a3ba845.png)

with my proposed patch we get the same, when allowing the space in it we would get:

![image](https://user-images.githubusercontent.com/5223533/144588913-40175483-0375-412d-ae44-51f440d9d5d1.png)


So I'm not sure whether or not the current implementation isn't correct.

Reply:
hmm, the commonmark spec mentions a [shortcut reference link](https://spec.commonmark.org/0.30/#shortcut-reference-link). This is what the detached `[1]` is then interpreted as, so you are right if we use this spec as the reference.